bin/genEvalSpecializations.py

```diff
@@ -85,6 +93,13 @@
 namespace DenseAd {
 
 {% if numDerivs < 0 %}\
+/*!
+ * \\brief Represents a function evaluation and its derivatives w.r.t. a
+ *        run-time specified set of variables.
+ */
+template <class ValueT>
+class Evaluation<ValueT, -1>
```

Comment:
I think that the use of -1 to signal runtime-size is fine (it's also used by Eigen I believe), but it should have a name, such as `enum { DynamicSize = -1 };` used throughout the script (and generated code).